internal/elasticsearch/index/ilm_test.go

```diff
@@ -273,12 +274,13 @@ resource "elasticstack_elasticsearch_index_lifecycle" "test_rollover" {
       max_age = "7d"
       max_docs = 10000
       max_size = "100gb"
+      max_primary_shard_docs = 5000
```

Comment:
We'll want to update the tests to only specify this when it's supported by the backing cluster. That means adding a case covering this attribute which is skipped when it's not supported ([example](https://github.com/elastic/terraform-provider-elasticstack/blob/main/internal/elasticsearch/index/ilm_test.go#L66-L79))